test/python/algorithms/test_grover.py

```diff
@@ -257,10 +257,15 @@ def test_circuit_result(self, use_sampler):
         if use_sampler:
             for i, dist in enumerate(result.circuit_results):
                 keys, values = zip(*sorted(dist.items()))
-                self.assertTupleEqual(keys, ("00", "01", "10", "11"))
                 if i in (0, 3):
-                    np.testing.assert_allclose(values, [0, 0, 0, 1], atol=0.2)
+                    if use_sampler == "ideal":
+                        self.assertTupleEqual(keys, ("00", "01", "10", "11"))
+                        np.testing.assert_allclose(values, [0, 0, 0, 1], atol=0.2)
```

Comment:
Now that 0 probabilities are not included, shouldn't this just show the "11" key here? That also seems to be what the error is about 🙂

Reply:
Ideal case should include "00", "01", and "10" with very small probabilities. The test on Linux passes but that on Mac fails. I'm investigating the difference.

Reply:
I cannot reproduce the CI failure locally on MacOS with Python 3.7 🤔 

The values with which "00", "01" and "10" are included are about 1e-34 maybe we don't really need to check that this is recognized as non-zero? This seems a bit volatile to me... since this is about algorithm performance rather than the technicality which element is populated and from the sampler tests we know that the zero-probability states are removed, how about just checking if the values are correct in the Grover test? E.g. like
```python
if i in (0, 3):
    values = [dist.get(key, 0) for key in ["00", "01", "10", "11"]]
    np.testing.assert_allclose(values, [0, 0, 0, 1], atol=0.2)
```

Reply:
I makes sense to me to filter out probabilities like this:

For the exact distribution: Remove probabilities satisfying $p_i \ll  1/|p| $. The reasoning is that the sum of any set of $p_i$ s of this magnitude is negligible. On the other hand we may want  to just remove what we think is in some sense a numerical artifact. This might not be easy to determine. For example `1e-34` is probably safe to remove.  But the former at least has a reason behind the choice. You still have to decide what is much less than one. Maybe `1e-10`

For the distribution from samples, you should remove a probability if $p_i / n_\text{shots} \ll 1$. This will make sampling efficient.

Reply:
I will use decimals of `Statevector.probabilities` to round the results.